track/align_guidescope.py

```diff
@@ -0,0 +1,145 @@
+import matplotlib.pyplot as plt
+import track
+from track import cameras
+from track.config import CONFIG_PATH
+from track.mounts import MeridianSide
+from track.model import (
+    save_default_param_set,
+    load_stored_param_set,
+    apply_guide_cam_alignment_error
+)
+
+
+SHUTDOWN_TIMEOUT_S = 2.0
+
+
+# FIXME: To shut down the subprocess cleanly, we need to send it SIGINT once and *only* once. Since
+# the subprocess is part of the same process group, when we do CTRL-C at the terminal *all*
+# processes in that group will receive SIGINT, and thus there is no need to send SIGINT to the
+# subprocess a second time under these conditions (and in fact problems arise if it gets SIGINT
+# more than once for reasons I don't fully understand). However, when this program ends for reasons
+# other than receiving CTRL-C, we *do* need to send SIGINT to the subprocess.
+
+
+def terminate_subprocess(process: subprocess.Popen) -> None:
+    """Terminate a running subprocess"""
+    print('Terminating subprocess...', end='', flush=True)
+    process.send_signal(signal.SIGINT)
+    time_sigint = time.perf_counter()
+    while process.poll() is None:
+        if time.perf_counter() - time_sigint > SHUTDOWN_TIMEOUT_S:
+            print('Subprocess shutdown timeout; resorting to SIGKILL...', end='', flush=True)
+            process.kill()
+            break
+        time.sleep(0.1)
+    print('done.')
+
+
+def main():
+    """See module docstring at the top of this file."""
+    parser = track.ArgParser()
+    parser.add_argument('star', help='Name of bright star')
+    parser.add_argument(
+        '--main-camera-cfg',
+        help='Configuration file with settings for the main OTA camera',
+        default=os.path.join(CONFIG_PATH, 'camera_main.cfg')
+    )
+    parser.add_argument(
+        '--meridian-side',
+        help='side of meridian for equatorial mounts to prefer',
+        required=True,
+        choices=tuple(m.name.lower() for m in MeridianSide),
+    )
+    cameras.add_program_arguments(parser, profile='align')
+    args = parser.parse_args()
+
+    # Run the `track` program as a subprocess. The intent is for this process to point the mount at
+    # the bright star and perform a spiral search until the star is found and centered in the main
+    # OTA camera.
+    track_process = subprocess.Popen(
+        args=[
+            'track',
+            f'--cfg={args.main_camera_cfg}',  # use the main OTA camera in this subprocess
+            f'--meridian-side={args.meridian_side}',
+            '--fuse',
+            '--spiral-search',
+            'star',
+            args.star,
+        ]
+    )
+    atexit.register(terminate_subprocess, track_process)
+
+    # prompt the user to ask when the track program has successfully found the star and centered it
+    # in the main OTA camera preview window
+    if not click.confirm(
+            'Is the star centered in the main OTA camera preview window?',
+            default=True
+        ):
```

Comment:
- [ ] I think this prompt shows up immediately after the spiral search starts. Should the prompt make it more obvious that the user may need to wait a bit for the search to find the star? E.g., "Started searching for [star name]. This might take a couple of minutes. Is the star centered in the camera preview window?"